crates/anvil/core/src/eth/transaction/mod.rs

```diff
@@ -1564,7 +1564,7 @@ impl Decodable for TypedReceipt {
 impl Encodable2718 for TypedReceipt {
     fn type_flag(&self) -> Option<u8> {
         match self {
-            Self::Legacy(_) => None,
+            Self::Legacy(_) => Some(0),
```

Comment:
Good catch, reverted this change

Reply:
yeah, it's just a bit different in rlp

though it is bad that we didn't catch this in tests as I believe this would result in invalid receipts root -> block hash